### ISSUE TITLE
Raycaster issues with intersection

### ISSUE THREAD

**Issue by aurbinaUCF** [aframevr/aframe#2501]
**Description:**

When trying to create a raycaster, it refuses to emit the event that it intersected with an object. It seems to only work if I were to wrap sphere around box as well but then the far attribute of raycaster does nothing and continues to maintian infinite length. Just looking for a way for an event to trigger if box intersects sphere.

- A-Frame Version: 0.5.0
- Platform / Device:
- Reproducible Code Snippet or URL:
`<script>
    AFRAME.registerComponent('collider-check', {
        dependencies: ['raycaster'],
        init: function () {
            this.el.addEventListener('raycaster-intersected', function () {
            console.log('Player hit something!');
            });
        }
    });
  </script>`
`<a-scene>
        <a-entity position="0 0 4">
            <a-entity id="cam" camera></a-entity>
        </a-entity>

        <a-entity id="sphere" collider-check geometry="primitive: sphere; radius: .5" material="color: red" position="0 0 0">
            <a-entity raycaster="objects: .collidable" position="0 0 0" rotation="90 0 0"></a-entity>
        </a-entity>
        
        <a-entity class="collidable" geometry="primitive: box" material="color: blue; opacity: .3" position="-1 0 0" wasd-controls></a-entity>
    </a-scene>`

**Comment by ngokevin** [comment]
Thanks for filing. I wonder if it's an issue with the `collidable` entity not being initialized in time the raycaster does its `querySelector`, even though the raycaster should be listening for attached entities. What happens if you swap the order, or remove the `objects` check?

**Comment by aurbinaUCF** [comment]
I'm not entirely sure but what I found out was that the collider-check tag on the sphere created a raycaster with default settings on the sphere entity itself rather than using the parameters of the wrapped raycaster. I removed the wrapped ray caster and adjusted the raycaster on the sphere and it worked fine.

Sent from my iPhone

On Mar 17, 2017, at 2:53 PM, Kevin Ngo <notifications@github.com<mailto:notifications@github.com>> wrote:


Thanks for filing. I wonder if it's an issue with the collidable entity not being initialized in time the raycaster does its querySelector, even though the raycaster should be listening for attached entities. What happens if you swap the order, or remove the objects check?
